### PR TITLE
Issue 15695 enhancements containerfile

### PR DESCRIPTION
**Pull request by jdotcms**

### CODE REVIEW

File: dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java [hunk 1]
```diff
@@ -281,18 +263,63 @@ private boolean isValidContainerPath(final Folder folder) {
 		return assets;
 	}
 
-	private Collection<? extends Permissionable> findFolderAssetContainers(final User user, final boolean includeArchived,
-																		   final Map<String, Object> params, final String hostId,
-																		   final String inode,  final String identifier,
-																		   final String parent, final String orderByParam) throws DotSecurityException, DotDataException {
+	private void getPaginatedAssets(final int offset,
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java]: It'd be great to have some Javadoc in these methods and classes.

File: dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java [hunk 2]
```diff
@@ -281,18 +263,63 @@ private boolean isValidContainerPath(final Folder folder) {
...
+			}
+		} else {
+			for(int i=0;i<toReturn.size();i++) {
+
+				assets.add((Container) toReturn.get(i));
+			}
+		}
+	}
+
+	private Collection<? extends Permissionable> findFolderAssetContainers(final User user, final String hostId,
```
Comment (jcastro-dotcms) [inline dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java]: It'd be great to have some Javadoc in these methods and classes.

**Review by jcastro-dotcms — CHANGES_REQUESTED** [review]

File: dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java [hunk 3]
```diff
@@ -281,18 +263,63 @@ private boolean isValidContainerPath(final Folder folder) {
...
+				switch (orderByParam) {
+					case "title asc":
+						containers.sort(Comparator.comparing(Container::getTitle));
+					break;
+
+                    case "title desc":
+                        containers.sort(Comparator.comparing(Container::getTitle).reversed());
+                        break;
+
+					case "modDate asc":
```
Comment (nollymar) [inline dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java]: consider a case insensitive comparison (ie.: moddate, title, etc)

Comment (jdotcms) [reply dotCMS/src/main/java/com/dotmarketing/portlets/containers/business/ContainerFactoryImpl.java]: done

**Review by jgambarios — APPROVED** [review]

**Pull request merged**